### PR TITLE
Use MIN and MAX macros.

### PR DESCRIPTION
**Pull request by nmoinvaz**
Use `MIN` and `MAX` macros in a few instances.

### CODE REVIEW

**Comment by nmoinvaz** [comment]
Let’s trick the code coverage tool by putting all our source code on one line. 100% coverage.

File: deflate.c [hunk 1]
```diff
@@ -1470,17 +1467,15 @@ static block_state deflate_stored(deflate_state *s, int flush) {
...
             }
             memcpy(s->window + s->strstart, s->strm->next_in - used, used);
             s->strstart += used;
             s->insert += MIN(used, s->w_size - s->insert);
         }
         s->block_start = (int)s->strstart;
     }
-    if (s->high_water < s->strstart)
-        s->high_water = s->strstart;
+    s->high_water = MIN(s->high_water, s->strstart);
```
Comment (mtl1979) [inline deflate.c]: I think this should be MAX().

File: deflate.c [hunk 2]
```diff
@@ -1500,18 +1495,15 @@ static block_state deflate_stored(deflate_state *s, int flush) {
...
-        have = s->strm->avail_in;
+    have = MIN(have, s->strm->avail_in);
     if (have) {
         read_buf(s->strm, s->window + s->strstart, have);
         s->strstart += have;
         s->insert += MIN(have, s->w_size - s->insert);
     }
-    if (s->high_water < s->strstart)
-        s->high_water = s->strstart;
+    s->high_water = MIN(s->high_water, s->strstart);
```
Comment (mtl1979) [inline deflate.c]: I think this should be MAX().

File: deflate.c (hunk 1 above)
Comment (nmoinvaz) [reply deflate.c]: Good catch I appreciate it!

File: deflate.c (hunk 2 above)
Comment (nmoinvaz) [reply deflate.c]: Thank you!

File: deflate.c (hunk 1 above)
Comment (mtl1979) [reply deflate.c]: You know... I go through most of pull requests if I just understand what the code does... There is still a lot of code I don't understand enough to be able to review.

**Comment by Dead2** [comment]
Rebase needed

File: deflate_stored.c [hunk 3]
```diff
@@ -132,7 +132,7 @@ Z_INTERNAL block_state deflate_stored(deflate_state *s, int flush) {
         }
         s->block_start = (int)s->strstart;
     }
-    s->high_water = MIN(s->high_water, s->strstart);
+    s->high_water = MAX(s->high_water, s->strstart);
```
Comment (nmoinvaz) [inline deflate_stored.c:135]: FYI, these were incorrectly converted to `MIN` in https://github.com/zlib-ng/zlib-ng/commit/d4dd539d1c08ce0affb360120a9e4ac2396e734b. Now have been corrected.

**Comment by nmoinvaz** [comment]
Rebased.

**Pull request merged**